File: src/028/problem028.py
#!/usr/bin/env python
n = 1001

d = [i**2 for i in range(1, n + 1, 2)]
d += [1 + i**2 for i in range(2, n + 1, 2)]
d += [1 + i + i**2 for i in range(1, n, 2)]
d += [1 - i + i**2 for i in range(3, n + 1, 2)]

print(sum(d))

File: src/028/problem028.exs
#!/usr/bin/env elixir
defmodule Problem028 do
  def spiral_diag_sum(1), do: 1
  def spiral_diag_sum(2), do: 10
  def spiral_diag_sum(x) when is_integer(x) and x > 0, do: 4*x*x - 6*(x-1) + spiral_diag_sum(x-2)

  def solve do
    spiral_diag_sum(1001)
  end
end

IO.puts Problem028.solve
